app/domain/finders/content.rb

```diff
@@ -44,6 +35,29 @@ def initialize(filter)
     @sort_direction = filter.fetch(:sort_direction) || 'desc'
   end
 
+  def view
+    @view ||= Finders::SelectView.new(date_range).run
+  end
+
+  def results
+    view[:model_name].all
+      .joins("INNER JOIN dimensions_editions ON aggregations_search_#{view[:table_name]}.dimensions_edition_id = dimensions_editions.id")
+      .joins("INNER JOIN facts_editions ON dimensions_editions.id = facts_editions.dimensions_edition_id")
+      .merge(slice_editions)
+      .order(sanitized_order(@sort_attribute, @sort_direction))
+      .page(@page)
+      .per(@page_size)
+      .select(*aggregates)
+  end
+
+  def total_results
+    view[:model_name].all
+      .joins("INNER JOIN dimensions_editions ON aggregations_search_#{view[:table_name]}.dimensions_edition_id = dimensions_editions.id")
+      .joins("INNER JOIN facts_editions ON dimensions_editions.id = facts_editions.dimensions_edition_id")
+      .merge(slice_editions)
+      .length
```

Comment:
Not 100% if this is still the case, but if you use `length` it will cast to an array and perform a count in memory. I would use `count`.

Reply:
Thanks Pablo. [Fixed](https://github.com/alphagov/content-performance-manager/pull/1137/commits/b5c918aa9e32fc1e1b230cfaa8bf0c9a9f29c735)